pkg/collector/corechecks/net/ntp.go

```diff
@@ -126,6 +135,28 @@ func (c *ntpConfig) parse(data []byte, initData []byte, getLocalServers func() (
 	return nil
 }
 
+func getCloudProviderNTPHosts() []string {
```

Comment:
Let's move it to pkg/util/cloudprovider.go and use the same logic than DetectCloudProvider with a generic function instead of duplicating `if.